homeassistant/components/binary_sensor/netatmo.py

```diff
@@ -0,0 +1,158 @@
+"""
+Support for the Netatmo binary sensors.
+
+The binary sensors based on events seen by the NetatmoCamera
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor.netatmo/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.util import Throttle
+from homeassistant.components.binary_sensor import (
+    BinarySensorDevice, PLATFORM_SCHEMA)
+from homeassistant.loader import get_component
+from homeassistant.const import CONF_MONITORED_CONDITIONS
+from homeassistant.helpers import config_validation as cv
+
+DEPENDENCIES = ["netatmo"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+# These are the available sensors mapped to binary_sensor class
+SENSOR_TYPES = {
+    "Someone known": "motion",
+    "Someone unknown": "motion",
+    "Motion": "motion",
+}
+
+CONF_HOME = 'home'
+CONF_CAMERAS = 'cameras'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_HOME): cv.string,
+    vol.Optional(CONF_CAMERAS, default=[]):
+        vol.All(cv.ensure_list, [cv.string]),
+    vol.Optional(CONF_MONITORED_CONDITIONS, default=SENSOR_TYPES.keys()):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+})
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=10)
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup access to Netatmo binary sensor."""
+    netatmo = get_component('netatmo')
+    home = config.get(CONF_HOME, None)
+
+    import lnetatmo
+    try:
+        data = WelcomeData(netatmo.NETATMO_AUTH, home)
+    except lnetatmo.NoDevice:
+        return None
+
+    if data.get_camera_names() == []:
+        return None
+
+    sensors = config.get(CONF_MONITORED_CONDITIONS, SENSOR_TYPES)
+
+    for camera_name in data.get_camera_names():
+        if config[CONF_CAMERAS] != []:
+            if camera_name not in config[CONF_CAMERAS]:
+                continue
+        for variable in sensors:
+            add_devices([WelcomeBinarySensor(data, camera_name, home,
+                                             variable)])
+
+
+class WelcomeBinarySensor(BinarySensorDevice):
+    """Represent a single binary sensor in a Netatmo Welcome device."""
+
+    def __init__(self, data, camera_name, home, sensor):
+        """Setup for access to the Netatmo camera events."""
+        self._data = data
+        self._camera_name = camera_name
+        self._home = home
+        if home:
+            self._name = home + ' / ' + camera_name
+        else:
+            self._name = camera_name
+        self._sensor_name = sensor
+        self._name += ' ' + sensor
+        self._unique_id = "Welcome_binary_sensor {}".format(self._name)
+        self.update()
+
+    @property
+    def name(self):
+        """The name of the Netatmo device and this sensor."""
+        return self._name
+
+    @property
+    def unique_id(self):
```

Comment:
Nah, if a platform implements it it will be better. They know exactly which is unique. However, a name is not very unique? Do you have access to a serial?